src/PageIterator.php

```diff
@@ -71,7 +72,7 @@ class PageIterator implements Page, \ArrayAccess, \JsonSerializable
      * @param mixed[] $parameters
      * @param array[] $columnDescriptors
      */
-    public function __construct(ResultIterator $parentResult, string $magicSql, array $parameters, int $limit, int $offset, array $columnDescriptors, ObjectStorageInterface $objectStorage, ?string $className, TDBMService $tdbmService, MagicQuery $magicQuery, int $mode, LoggerInterface $logger)
+    public function __construct(ResultIterator $parentResult, ?string $magicSql, array $parameters, ?int $limit, ?int $offset, ?array $columnDescriptors, ?ObjectStorageInterface $objectStorage, ?string $className, ?TDBMService $tdbmService, ?MagicQuery $magicQuery, ?int $mode, ?LoggerInterface $logger)
```

Comment:
Same comment as above. We could put the constructor private and empty, and create 2 static methods to create an instance.

Reply:
done